sdk/tests/conformance2/context/context-attributes-depth-stencil-antialias-obeyed.html

```diff
@@ -0,0 +1,296 @@
+
+    // We can't use wtu.create3DContext because it defaults to antialias=false.
+    var names = ["webgl2", "experimental-webgl2"]; 
+    for (var i = 0; i < names.length; ++i) {
+      try {
+        gl = canvas.getContext(names[i], attribs);
+      } catch (e) {
+      }
+      if (gl) {
+        break;
+      }
+    }
+    if (!gl)
+        return null;
+
+    var program = wtu.setupProgram(gl, ["vshader", "fshader"], ["pos", "colorIn"]);
+    if (!program)
+        return null;
+
+    gl.enable(gl.DEPTH_TEST);
+    gl.enable(gl.STENCIL_TEST);
+
+    gl.clearColor(clearColor[0], clearColor[1], clearColor[2], clearColor[3]);
+    gl.clearDepth(clearDepth);
+    gl.clearStencil(clearStencil);
+    gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT | gl.STENCIL_BUFFER_BIT);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
+    return gl;
+}
+
+function draw(gl, vertices, colors)
+{
+    var colorOffset = vertices.byteLength;
+
+    var vbo = gl.createBuffer();
+    gl.bindBuffer(gl.ARRAY_BUFFER, vbo);
+    gl.bufferData(gl.ARRAY_BUFFER, colorOffset + colors.byteLength, gl.STATIC_DRAW);
+    gl.bufferSubData(gl.ARRAY_BUFFER, 0, vertices);
+    gl.bufferSubData(gl.ARRAY_BUFFER, colorOffset, colors);
+
+    gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 0, 0);
+    gl.enableVertexAttribArray(0);
+    gl.vertexAttribPointer(1, 4, gl.UNSIGNED_BYTE, true, 0, colorOffset);
+    gl.enableVertexAttribArray(1);
+
+    gl.drawArrays(gl.TRIANGLES, 0, vertices.length / 3);
+}
+
+function testDefault()
+{
+    debug("");
+    debug("Testing default attributes: { antialias: true, depth: true, stencil:false }");
+    shouldBeNonNull("gl = getWebGL(1, 1, {}, [ 0, 0, 0, 0 ], 1, 0)");
+    shouldBeTrue("gl.getContextAttributes().antialias");
+    shouldBeTrue("gl.getContextAttributes().depth");
+    shouldBeTrue("gl.getParameter(gl.DEPTH_BITS) >= 16");
+    shouldBeFalse("gl.getContextAttributes().stencil");
+    shouldBeTrue("gl.getParameter(gl.STENCIL_BITS) == 0");
+}
+
+function testDepth(depth)
+{
+    debug("");
+    debug("Testing depth = " + depth);
+    if (depth) {
+        shouldBeNonNull("gl = getWebGL(1, 1, { stencil: false, antialias: false }, [ 0, 0, 0, 1 ], 1, 0)");
+        shouldBeTrue("gl.getParameter(gl.DEPTH_BITS) >= 16");
+    } else {
+        shouldBeNonNull("gl = getWebGL(1, 1, { depth: false, stencil: false, antialias: false }, [ 0, 0, 0, 1 ], 1, 0)");
+        shouldBeTrue("gl.getParameter(gl.DEPTH_BITS) == 0");
+    }
+    shouldBeTrue("gl.getParameter(gl.RED_BITS) >= 8");
+    shouldBeTrue("gl.getParameter(gl.GREEN_BITS) >= 8");
+    shouldBeTrue("gl.getParameter(gl.BLUE_BITS) >= 8");
+    shouldBeTrue("gl.getParameter(gl.ALPHA_BITS) >= 8");
+
+    shouldBeNonNull("contextAttribs = gl.getContextAttributes()");
+    if (depth != contextAttribs.depth) {
+        testFailed("WebGL implementation doesn't obey depth attribute");
+    } else {
+        testPassed("WebGL implementation obeys depth attribute");
+    }
```

Comment:
Since it looks like the various checks ensuring depth, stencil and antialias are honored are the only difference compared to conformance/context/context-attributes-alpha-depth-stencil-antialias.html, do you think it's worth refactoring the test into a separate .js file so it can be shared between the WebGL 1.0 and 2.0 versions of the test?